workflows/multiomics/full_pipeline/main.nf

```diff
@@ -62,11 +63,26 @@ workflow run_wf {
 
     modality_processors = [
       [id: "rna", singlesample: rna_singlesample, multisample: rna_multisample],
-      [id: "vdj_t", singlesample: null, multisample: null],
-      [id: "vdj_b", singlesample: null, multisample: null],
-      [id: "prot", singlesample: null, multisample: null],
-      [id: "atac", singlesample: null, multisample: null],
+      // [id: "vdj_t", singlesample: null, multisample: null],
+      // [id: "vdj_b", singlesample: null, multisample: null],
+      // [id: "prot", singlesample: null, multisample: null],
+      // [id: "atac", singlesample: null, multisample: null],
     ]
+    known_modalities = modality_processors.collect{it.id}
+
+    unknown_channel = start_ch
+      | filter { ! known_modalities.contains(it[2].modality)}
+      | map { lst ->
+          [lst[2].modality] + lst
+       }
+      | groupTuple(by: 0)
+      // [modality, [sample_id1, sample_id2, ...], [h5mu1, h5mu2, ...], [passthrough, copy_passthrough, ...]]
+      | map { grouped_lst ->
+        modality_name = grouped_lst[0]
```

Comment:
use `def modality_name = ...` to ensure that the variable doesn't leak out of its environment